stdlib/public/core/StringLegacy.swift

```diff
@@ -124,24 +124,20 @@ extension String {
     if prefixCount == 0 {
       return true
     }
-    if let unmanagedSelf = self._guts._unmanagedContiguous,
-      let unmanagedPrefix = prefix._guts._unmanagedContiguous {
-      defer {
-        _fixLifetime(self)
-        _fixLifetime(prefix)
-      }
-      if unmanagedSelf.isSingleByte, unmanagedPrefix.isSingleByte {
+    if _fastPath(!self._guts._isOpaque && !prefix._guts._isOpaque) {
+      defer { _fixLifetime(self); _fixLifetime(prefix) }
```

Comment:
Complex closures like this can also incur extra ARC traffic. They should be very simple to sink.